File: examples/java/echo/src/main/java/ftl/echo/Echo.java
package ftl.echo;

import ftl.time.TimeClient;
import xyz.block.ftl.Export;
import xyz.block.ftl.FunctionVerb;
import xyz.block.ftl.Verb;

@Export
@Verb
public class Echo implements FunctionVerb<EchoRequest, EchoResponse> {

    final TimeClient timeClient;

    public Echo(TimeClient timeClient) {
        this.timeClient = timeClient;
    }

    public EchoResponse call(EchoRequest req) {
        var response = timeClient.call();
        return new EchoResponse("Hello, " + req.name().orElse("anonymous") + "! The time is " + response.getTime() + ".");
    }
}

File: examples/kotlin/echo/src/main/kotlin/ftl/echo/Echo.kt
package ftl.echo

import ftl.time.TimeClient
import xyz.block.ftl.Export
import xyz.block.ftl.FunctionVerb
import xyz.block.ftl.Verb

data class EchoRequest(val name: String?)
data class EchoResponse(val message: String)

@Export
@Verb
class Echo (val time: TimeClient): FunctionVerb<EchoRequest, EchoResponse> {
  override fun call(req: EchoRequest): EchoResponse {
    val response = time.call()
    return EchoResponse(message = "Hello, ${req.name ?: "anonymous"}! The time is ${response.time}.")
  }
}
